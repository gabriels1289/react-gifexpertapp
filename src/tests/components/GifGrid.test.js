import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas con GifGrid', () => {

  const category = 'One piece';

  test('debe de mostrarse correctamente', () => {    
    
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow( <GifGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
  });

  test('debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

    const gifs = [{
      id: 'abc',
      url: 'https://loclahost/cualquier/cosa.jpg',
      title: 'Cualquier cosa',
    },
    {
      id: '123',
      url: 'https://loclahost/cualquier/cosa.jpg',
      title: 'Cualquier cosa',
    }
    ];

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    
    const wrapper = shallow( <GifGrid category={category} /> );
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe( false );
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );

  });

})