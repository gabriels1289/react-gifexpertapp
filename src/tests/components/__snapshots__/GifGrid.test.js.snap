// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con GifGrid debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="abc"
      key="abc"
      title="Cualquier cosa"
      url="https://loclahost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://loclahost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas con GifGrid debe de mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    One piece
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
